### ISSUE TITLE
Implement header navigation component

### ISSUE THREAD

**Issue by ghost** [ocaml/v3.ocaml.org-rescript#125]
- [x] Implement non-interactive menu for mobile and large screens
- [x] Implement a minimal interactive menu for large screens
- [x] Implement a minimal, interactive menu for mobile screens
- [ ] Update to match current large screen and mobile screen Figma designs (could be a few hours or two days, depending on complexity of CSS)
- [x] Change large screen menu to use `hidden` instead of `opacity` to fix subtle issue with hidden menu capturing mouse events
- [ ] Change large screen menu to use `z-order` to ensure dropdown menus always appear above other content

**Comment by ghost** [comment]
> * Implement a simple, interactive menu for large screens

Done in #160

**Comment by ghost** [comment]
> * Implement non-interactive menu for mobile and large screens

This was done in #86

**Comment by ghost** [comment]
>  Implement a simple, interactive menu for mobile screens

This step will implement the logic to show and hide the mobile hamburger menu. #172 will implement the content that resides inside of the dropdown menu, once this logic is in place.

**Comment by ghost** [comment]
Tailwind didn't indicate this clearly, but I needed to make use of both `z-order` and `hidden` classes to cause proper showing and hiding.

**Comment by ghost** [comment]
> * Implement a minimal, interactive menu for mobile screens

Completed in #209

**Comment by ghost** [comment]
> * Change large screen menu to use hidden instead of opacity to fix subtle issue with hidden menu capturing mouse events

[Done within this PR](../pull/212/files#diff-ce8d99adea048ae66fcf9409fa894eccbf1d8c5e0a1b2b7d9677770314e962c4)

**Comment by ghost** [comment]
> Attempt refactoring implementation to make use of headless-ui library (timebox this, create a follow-up issue if it turns out to be complex)

This appears to be partially or fully done in PR #385.
